### ISSUE TITLE
rustc hangs on `dotenv!` macro

### ISSUE THREAD

**Issue by mammothbane** [purpliminal/rust-dotenv#79]
I'm experiencing compiler hangs if I use the `dotenv!` macro anywhere on recent nightlies. [Here](https://gist.github.com/mammothbane/0f7a53492c8fc70c90c7b87a2e212d7c) is a gist that reproduces the problem. This has occurred on Windows 10 machines running:

- `nightly-x86_64-pc-windows-gnu 1.25.0 16362c737`
- `nightly-x86_64-pc-windows-gnu 1.25.0 4d2d3fc5d`

and `-msvc` builds at the same commits. Also confirmed on Ubuntu 17.10 nightly at `4d2d3fc5d`.

This is my output from `cargo rustc -- -Ztime-passes` (run after deps are built and I've already gotten a hang):

```
E:\code\dotenv_breaks> cargo rustc -- -Ztime-passes
   Compiling dotenv_breaks v0.1.0 (file:///E:/code/dotenv_breaks)
  time: 0.000; rss: 20MB        parsing
  time: 0.000; rss: 20MB        recursion limit
  time: 0.000; rss: 20MB        crate injection
  time: 0.000; rss: 20MB        plugin loading
  time: 0.000; rss: 20MB        plugin registration
  time: 0.000; rss: 20MB        background load prev dep-graph
```

And it hangs right there indefinitely, though it still uses some CPU (30-40%) but not much memory.

It will *also* hang if I remove the key from the `.env` file, though it panics first.

**Comment by mammothbane** [comment]
After some more testing: I'm also getting the same hangs on stable (1.23.0).

**Comment by mammothbane** [comment]
resolved by #91
